login.html

```diff
@@ -0,0 +1,59 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Login Now</title>
+    <script src="/main.js"></script>
+    <script type='text/javascript' src="/config.js"></script>
+    <script src="./login-init.js"></script>
+</head>
+
+<body>
+    <h1>Login Now</h1>
+
+    <div id="firebaseui-auth-container"></div>
+    <div id="loader">Loading...</div>
+
+    <input id="loginButton" type="button" value="Login with Google" onclick="login();" />
+    <input id="logoutButton" type="button" value="Logout" onclick="logout();" />
+
+    <!-- Firebase with CDN-->
+    <!-- Firebase App (the core Firebase SDK) is always required and must be listed first -->
+    <script src="https://www.gstatic.com/firebasejs/8.3.0/firebase-app.js"></script>
+
+    <!-- If you enabled Analytics in your project, add the Firebase SDK for Analytics -->
+    <script src="https://www.gstatic.com/firebasejs/8.3.0/firebase-analytics.js"></script>
+
+    <!-- Add Firebase products that you want to use -->
+    <script src="https://www.gstatic.com/firebasejs/8.3.0/firebase-auth.js"></script>
+    <script src="https://www.gstatic.com/firebasejs/8.3.0/firebase-firestore.js"></script>
+
+    <script src="https://www.gstatic.com/firebasejs/ui/4.6.1/firebase-ui-auth.js"></script>
+    <link type="text/css" rel="stylesheet" href="https://www.gstatic.com/firebasejs/ui/4.6.1/firebase-ui-auth.css" />
+    <script>
+        // For Firebase JavaScript SDK v7.20.0 and later, `measurementId` is an optional field
+        var firebaseConfig = {
```

Comment:
This stanza appears in `popup.html` too. Why? The *don't repeat yourself* (DRY) principle is generally preferable so there should be a reason why you have ignored it. Perhaps some refactoring is needed?

Reply:
I need some help here. How would you refactor it? What's the best approach to do so?

Reply:
Add a function (for example) like:

```
function makeFirebaseConfig() {
  return {
            apiKey: config().apiKey,
             authDomain: config().authDomain,
             projectId: config().projectId,
             storageBucket: config().storageBucket,
             messagingSenderId: config().messagingSenderId,
             appId: config().appId,
             measurementId: config().measurementId
    };
}
```

to one of the JavaScript files included above in both places. Then change the init line to:

```
firebase.initializeApp(makeFirebaseConfig);
```

Reply:
ahhhhhhhh I thought you meant to refactor the scripts tags hahahah 😆

Reply:
Awesome. Sorry that this was unclear.